gateway/ln-gateway/src/lib.rs

```diff
@@ -419,113 +425,148 @@ impl Gateway {
-                                        password: None,
-                                        network: Some(lightning_network),
-                                        num_route_hints: None,
-                                        routing_fees: None,
-                                        per_federation_routing_fees: None,
-                                    }).await.expect("Failed to set gateway configuration");
-                                    continue;
-                                }
+                    let htlc_task_group = tg.make_subgroup();
+                    let lnrpc_route = self_copy.lightning_builder.build().await;
 
-                                info!("Successfully loaded Gateway clients.");
-                                let lightning_context = LightningContext {
-                                    lnrpc: ln_client,
-                                    lightning_public_key,
-                                    lightning_alias,
-                                    lightning_network,
-                                };
-                                self_copy.set_gateway_state(GatewayState::Running {
-                                    lightning_context
-                                }).await;
-
-                                // Blocks until the connection to the lightning node breaks or we receive the shutdown signal
-                                if handle.cancel_on_shutdown(self_copy.handle_htlc_stream(stream, handle.clone())).await.is_ok() {
-                                    warn!("HTLC Stream Lightning connection broken. Gateway is disconnected");
-                                } else {
-                                    info!("Received shutdown signal");
-                                    self_copy.handle_disconnect(htlc_task_group).await;
-                                    break;
-                                }
-                            }
-                            Err(e) => {
-                                warn!("Failed to retrieve Lightning info: {e:?}");
-                            }
+                    debug!("Establishing HTLC stream...");
+                    let (stream, ln_client) = match lnrpc_route.route_htlcs(&htlc_task_group).await
+                    {
+                        Ok((stream, ln_client)) => (stream, ln_client),
+                        Err(e) => {
+                            warn!("Failed to open HTLC stream: {e:?}");
+                            continue
                         }
+                    };
+
+                    // Successful calls to route_htlcs establish a connection
+                    self_copy.set_gateway_state(GatewayState::Connected).await;
+                    info!("Established HTLC stream");
+
+                    let route_htlcs_response =
+                        self_copy.route_htlcs(&handle, stream, ln_client).await;
```

Comment:
Let's take this opportunity to improve on our naming to support LNv2. `route_htlcs` is a bad name since its the same as the function above. Can we call this `handle_payment_requests` and rename `route_htlcs` to `create_payment_stream`? (open to suggestions on naming, but in general we need to move away from using "HTLC")

Reply:
Although, maybe we want to merge this first so the diff isn't impossible to read

Reply:
Yeah, I think it makes sense to handle this in a separate PR. Let's keep the "htlc" naming for now for consistency.